PowerShellGet/PSModule.psm1

```diff
@@ -4906,6 +4906,16 @@ function Test-ScriptFileInfo
             return
         }
 
+        if ($PSScriptInfo.Version -match '-')
+        {
+            # Version contains PreRelease string.  Break apart and validate separately.
+            $indexOfFirstHyphen = $PSScriptInfo.Version.IndexOf('-')
+            $versionSubstring = $PSScriptInfo.Version.Substring(0, $indexOfFirstHyphen)
+            $preReleaseSubstring = $PSScriptInfo.Version.Substring($indexOfFirstHyphen+1)
+
+            $preReleaseIsValid = Validate-PreReleaseString -Version $versionSubstring -PreRelease $preReleaseSubstring -InfoObject $PSScriptInfo

```

Comment:
There are a couple places where I perform an if-condition on Validate-PrereleaseString.  Will the if-conditions still work even when it does not return a true/false?